metadata/src/main/java/org/apache/kafka/metadata/placement/TopicAssignment.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.metadata.placement;
+
+import java.util.List;
+
+/**
+ * The KRaft topic assignment.
+ */
+public class TopicAssignment {
+    private List<PartitionAssignment> assignments;
+
+    public TopicAssignment(List<PartitionAssignment> assignments) {
+        this.assignments = assignments;
```

Comment:
We should make this type immutable.

Reply:
I can update this to:
```
    public TopicAssignment(final List<PartitionAssignment> assignments) {
        this.assignments = Collections.unmodifiableList(assignments);
    }
```
Does that make sense?

Reply:
Yes. Ideally you would also clone/copy the parameter pass in the constructor because the caller has a reference to `assignments` and can update it.

Reply:
Updated to:
```
    public TopicAssignment(final List<PartitionAssignment> assignments) {
        this.assignments = Collections.unmodifiableList(assignments);
    }
```
Also commented on both this class and `PartitionAssignment` that they're immutable.